### DISCUSSION TITLE
Q: are multiple alarms possible with this board?

### DISCUSSION THREAD

**Discussion by BjornW** [joeycastillo/Sensor-Watch#75] [Q&A]
I've always wanted to have the option to create multiple alarms in my Casio F-91W. Is this possible with this board? If so how many? Would be great if I could set ~5 different alarms.

**Comment by joshber** [comment]
I’ve been thinking along similar lines. It might entail building with multiple copies of the alarm face so that they can each bind a wants_background_task() handler.

Here’s an alarm face I roughed in today (from my clone of the repo):

https://bitbucket.org/thylacine/sensor-watch_clone/src/main/movement/watch_faces/complication/wake_face.c
https://bitbucket.org/thylacine/sensor-watch_clone/src/main/movement/watch_faces/complication/wake_face.h

I’ve used wants_background_task() to ensure the event will trigger even if the runtime is in sleep mode. A further concern is that for accessibility I’d like to offer an LED signal as well as piezo. At the moment I’m spawning a thread that turns on the LED, goes to sleep, and turns it off some seconds later — so that I’m not blocking the event loop thread. Not sure if this approach is compatible with this module, need to test it in the emulator.

**Comment by tahnok** [comment]
Nice, your approach sounds good in practice, although I wonder if it's
better just to bring the alarm watch face to the foreground (I think
there's a method for that)

I'd suggest you put up a PR for a basic alarm first, then multiple alarms
(maybe multiple copies of the watch face? I think that's been done for the
days since face before)

**Reply by joshber** [reply]
🙇 just want to make sure it builds and the behavior looks reasonable in the emulator. I’ll try to get a PR in by the end of the week.

Bringing the face to the foreground when the callback triggers is an interesting thought. I always hate it when an application takes the userland focus when it’s finished starting up or something. And I could imagine scenarios where you’re checking the temp log, say, and the wake-up gets in your face. It would make sense if, say, you could “Dismiss the alarm just for the current cycle” (maybe a long press on the adjust button, which sets a flag that shows up in the UI as the LAP indicator)?
